pkgs/development/compilers/ghc/8.10.7.nix

```diff
@@ -400,7 +432,8 @@ stdenv.mkDerivation (rec {
       guibou
     ] ++ lib.teams.haskell.members;
     timeout = 24 * 3600;
-    inherit (ghc.meta) license platforms;
+    platforms = lib.platforms.all;
```

Comment:
This change is right, but means we need to find a new mechanism to implement this: https://github.com/NixOS/nixpkgs/pull/240348. Not sure how that should look tbh, there is not a lot of prior art on detecting if something can be boostrapped in nixpkgs…

Reply:
Ideally, I think we'd want a way of recursively checking that the entire closure supports the build platform and isn't marked broken. I'm not aware of an existing function to check that.

Perhaps it'd be easier to give users the option to override `writeShellApplication` to disable the usage of ShellCheck.

If we still want to automatically choose whether to use it or not, how about this?

1. Check if ShellCheck is marked broken.
2. Check if the GHC used for ShellCheck is marked broken or unavailable.
3. Check if the boot GHC used is marked broken or unavailable (repeat until we get to a pre-compiled GHC).

It's not general enough to be usable on anything in Nixpkgs, but it should at the very least prevent a regression here.